pkg/api/validation/validation.go

```diff
@@ -800,6 +800,22 @@ func ValidateLimitRange(limitRange *api.LimitRange) errs.ValidationErrorList {
 	return allErrs
 }
 
+// ValidateSecret tests if required fields in the Secret are set.
+func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(secret.Name) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("name", secret.Name))
+	} else if !util.IsDNSSubdomain(secret.Name) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", secret.Name, ""))
+	}
+	if len(secret.Namespace) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("namespace", secret.Namespace))
+	} else if !util.IsDNSSubdomain(secret.Namespace) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", secret.Namespace, ""))
+	}
+	return allErrs
```

Comment:
Yes. 

More precisely, you might say you want to support up to 256 pods with 4 secrets each, and maximum 64kB per secret, so you need to set aside 64MB memory for that purpose.  Just making up values.

Also, instead of a tmpfs per secret, you also might do a single tmpfs per node and then bind mount subtrees of the tmpfs into the appropriate containers.    Pretty sure you hit some kind of limit on some versions of linux if you try to make 1024 tmpfs mounts.